src/shared/pages/workstack.jsx

```diff
@@ -79,15 +79,20 @@ class WorkstackPage extends Component {
         // TODO: Remove
         /* eslint-disable-next-line no-undef */
         const payload = new FormData();
-        Object.keys(formData).forEach(field => {
-            if (Array.isArray(formData[field])) {
-                formData[field].map(value => {
-                    payload.append(`${field}`, value);
+
+        for (const [key, value] of
+            Object.entries(formData)
+                .filter(([_, value]) => (value !== null && value !== undefined))) {
+            if (Array.isArray(value)) {
+                value.map(v => {
+                    payload.append(`${key}`, v);
                 });
             } else {
-                payload.append(field, formData[field]);
+                if (typeof value === 'string') {
+                    payload.append(key, value);
+                }
             }
-        });
+        }
 
```

Comment:
As this submit handler is doing the allocations, I wonder if we should dispatch a call to `updateApiStatus` in here (probably after the post) to state allocation has happened/failed. We could potentially throw it if the values aren't valid around this logic too potentially?

If we did that this could be extracted out of this handler method and specific tests written too for it, as now we're doing a bit more logic than the previous implementation.

Interested to get your thoughts on this?